### ISSUE TITLE
Release SecureDrop 2.0.2

### ISSUE THREAD

**Issue by zenmonkeykstop** [freedomofpress/securedrop#6055]
This is a tracking issue for the release of SecureDrop 2.0.2

Tentatively scheduled as follows:

**Pre-release announcement:** 2021-08-11
**Release date:** 2021-08-12

**Release manager:** KOG
**Deputy release manager:** n/a
**Communications manager:**: EM

_SecureDrop maintainers and testers:_ As you QA 2.0.2, please report back your testing results as comments on this ticket. File GitHub issues for any problems found, tag them "QA: Release", and associate them with the [2.0.2 milestone](https://github.com/freedomofpress/securedrop/milestone/75) for tracking (or ask a maintainer to do so).

Test debian packages will be posted on https://apt-test.freedom.press signed with [the test key](https://gist.githubusercontent.com/conorsch/ec4008b111bc3142fca522693f3cce7e/raw/2968621e8ad92db4505a31fcc5776422d7d26729/apt-test%2520apt%2520pubkey). An Ansible playbook testing the upgrade path is [here](https://TK/).

# [QA Matrix for 2.0.2](https://docs.google.com/spreadsheets/d/1ilHLJunjNSw57S4HcUYOB8TDtSAyMq6AXdPx8UfI4hM/edit#gid=1698153040)
# [Test Plan for 2.0.2](https://github.com/freedomofpress/securedrop/wiki/2.0.2-Test-Plan)

# Prepare release candidate (2.0.2~rc1)
- [x] Link to latest version of Tails, including release candidates, to test against during QA
- [x] Prepare 2.0.2~rc1 release changelog
- [x] Branch off release/2.0.2 from release/2.0.1 
- [x] Prepare 2.0.2~rc1
- [x] Build debs, preserving build log, and put up `2.0.2~rc1` on test apt server
- [x] Commit build log.

After each test, please update the QA matrix and post details for Basic Server Testing, Application Acceptance Testing and release-specific testing below in comments to this ticket.

# Final release
- [x] Ensure builder in release branch is updated and/or update builder image
- [x] Push signed tag 
- [x] Pre-Flight: Test updater logic in Tails (apt-qa tracks the `release` branch in the LFS repo)
- [x] Build final Debian packages for 2.0.2 (and preserve build log)
- [x] Commit package build log to https://github.com/freedomofpress/build-logs
- [x] Upload Debian packages, including `5.4.136` grsec kernel packages, to apt-qa server
- [x] Pre-Flight: Test that install and upgrade from 2.0.1 to 2.0.2 works w/ prod repo debs (apt-qa.freedom.press polls the `release` branch in the LFS repo for the debs)
- [x] Flip apt QA server to prod status (merge to `main` in the LFS repo)
- [x] Merge Docs branch changes to ``main`` and verify new docs build in securedrop-docs repo
- [x] Prepare release messaging

# Post release
- [x] Create GitHub release object 
- [x] Once release object is created, update versions in `securedrop-docs` and Wagtail
- [x] Build readthedocs and verify new docs show up on https://docs.securedrop.org
- [x] Publish announcements
- [x] Merge changelog back to `develop`
- [x] Update roadmap wiki page: https://github.com/freedomofpress/securedrop/wiki/Development-Roadmap

**Comment by zenmonkeykstop** [comment]
## 2.0.2 QA Checklist

For both upgrades and fresh installs, here is a list of functionality that requires testing. You can use this for copy/pasting into your QA report.

### Environment

- Install target: NUC10-app/NUC8-mon
- Tails version: 4.21
- Test Scenario: fresh install
- SSH over Tor: no
- Release candidate: 2.02-rc1
- General notes:

### Basic Server Testing
- [x] After installing the testinfra dependencies, all tests in `./securedrop-admin verify` are passing:
  - Install dependencies on Admin Workstation with `cd ~/Persistent/securedrop && ./securedrop-admin setup -t`
  - Run tests with `./securedrop-admin verify` (this will take a while)
  - Remove test dependencies: `rm -rf admin/.venv3/ && ./securedrop-admin setup`
- [x] QA Matrix checks pass

#### Command Line User Generation

- [x] Can successfully add admin user and login

#### Administration

- [x] I have backed up and successfully restored the app server following [the backup documentation](https://docs.securedrop.org/en/latest/backup_and_restore.html)
- [ ] If doing upgrade testing, make a backup on 2.0.1 and restore this backup on this release candidate **n/a**
- [x] "Send Test OSSEC Alert" button in the journalist triggers an OSSEC alert and an email is sent
- [ ] Can successfully add journalist account with HOTP authentication **not tested**

### Application Acceptance Testing

#### Source Interface

##### Landing page base cases
- [x] JS warning bar does not appear when using Security Slider high
- [x] JS warning bar does appear when using Security Slider Low

##### First submission base cases
- [x] On generate page, refreshing codename produces a new 7-word codename
- [x] On submit page, empty submissions produce flashed message
- [x] On submit page, short message submitted successfully
- [x] On submit page, file greater than 500 MB produces "The connection was reset" in Tor Browser _quickly_ before the entire file is uploaded
- [x] On submit page, file less than 500 MB submitted successfully

##### Returning source base cases

- [x] Nonexistent codename cannot log in
- [x] Empty codename cannot log in
- [x] Legitimate codename can log in
- [x] Returning user can view journalist replies - need to log into journalist interface to test

#### Journalist Interface 

##### Login base cases
- [x] Can log in with 2FA tokens
- [x] incorrect password cannot log in
- [x] invalid 2fa token cannot log in
- [x] 2fa immediate reuse cannot log in
- [ ] Journalist account with HOTP can log in

##### Index base cases

- [x] Filter by codename works
- [x] Starring and unstarring works
- [x] Click select all selects all submissions
- [x] Selecting all and clicking "Download" works

##### Individual source page

- [x] You can submit a reply and a flashed message and new row appears
- [x] You cannot submit an empty reply
- [x] Clicking "Delete Source Account" and the source and docs are deleted
- [x] You can click on a document and successfully decrypt using application private key

### Basic Tails Testing

After updating to this release candidate and running `securedrop-admin tailsconfig`

- [x] The Updater GUI appears on boot
- [x] Updating to 2.0.1 is successful

### 2.0.2 release-specific changes

- [x] Kernel is updated to 5.4.136 [#6054](https://github.com/freedomofpress/securedrop/pull/6054)
  - [x] both app and mon servers are using the 5.4.136 grsec kernel 
  - [x] paxtest tests return expected results

### Preflight testing

#### Basic testing

- [ ] Install or upgrade occurs without error
- [ ] Source interface is available and version string indicates it is 2.0.2
- [ ] A message can be successfully submitted

#### Tails

- [ ] The updater GUI appears on boot
- [ ] The update successfully occurs to 2.0.2
- [ ] After reboot, updater GUI no longer appears

**Comment by zenmonkeykstop** [comment]
Have also confirmed that upgrade path works for 1U servers and that 5.4.136 is present and correct.

**Comment by eloquence** [comment]
Release comms draft (minimal as it is) here: https://docs.google.com/document/d/1np9z-E9GZAn_HLliPzVC9kaPZOBVberuw7wZh9p3mto/edit#

**Comment by kushaldas** [comment]
Upgrade path works on NUC5.

**Comment by sssoleileraaa** [comment]
## 2.0.2 QA Checklist

For both upgrades and fresh installs, here is a list of functionality that requires testing. You can use this for copy/pasting into your QA report.

### Environment

- Install target: NUC7i5BNH-app/NUC7i5BNH-mon
- Tails version: 4.21
- Test Scenario: "Clean install scenario"
- SSH over Tor: no
- Release candidate: rc1
- General notes:

### Clean install
1. Check out rc1 tag
2. Edit `install_files/ansible_base/roles/install-fpf-repo/defaults/main.yml` to use:

```
apt_repo_url: https://apt-test.freedom.press
...
apt_repo_pubkey_files:
    - apt-test-signing-key.pub
```

3. To configure and install SecureDrop, run the following:
```
cd ~/Persistent/securedrop
./securedrop-admin --force setup
./securedrop-admin --force sdconfig
./securedrop-admin --force install
./securedrop-admin --force tailsconfig
```

4. Confirm package versions:
   * `apt-cache policy securedrop-app-code` -> Installed: 2.0.2~rc1+focal
   * `apt-cache policy securedrop-config` -> Installed: 0.1.4+2.0.2~rc1+focal
   * `apt-cache policy securedrop-keyring` -> Installed: 0.1.5+2.0.2~rc1+focal
   * `apt-cache policy securedrop-ossec-agent` -> Installed: 3.6.0+2.0.2~rc1+focal
   * `apt-cache policy securedrop-ossec-server` -> Installed: 3.6.0+2.0.0~rc1+focal (mon)
   * `apt-cache policy securedrop-grsec` -> Installed: 5.4.136+focal

### Kernel testing

- [x] `paxtest blackhat` has expected results  
   - `memcpy` and `memcpy, PIE` shows "Vulnerable" (see https://github.com/freedomofpress/securedrop/issues/1039#issuecomment-790033219)
   - `strcpy and `strcpy, PIE` show `paxtest: return address contains a NULL byte`
   - Randomization tests: bit guesses in the high 20s or higher (there may be one or two in the mid-20s, this is still okay).  
   - Everything else: "Killed"

#### Additional Kkernel testing on app server only
- [x] `curl -L https://meltdown.ovh -o meltdown.sh && sudo bash meltdown.sh -v`
   - `CVE-2018-365` aka 'Foreshadow (SGX), L1 terminal fault' shows *vulnerable* (see https://github.com/freedomofpress/securedrop/issues/5040#issuecomment-559597643)
   - Other tests do not show vulnerable

### Basic Server Testing

- [x] `./securedrop-admin --force verify` passes
  - Set up virtualenv with test dependencies: `cd ~/Persistent/securedrop && rm -r admin/.venv3 && ./securedrop-admin setup -t`
  - Run tests: `./securedrop-admin verify` (this will take a while)
  - Cleanup (remove test dependencies): `rm -rf admin/.venv3/ && ./securedrop-admin setup`
- [x] QA Matrix checks pass
- [x] I can access both the source and journalist interfaces
- [x] I can SSH into both machines over Tor
- [x] AppArmor is loaded on app
    - [x] 0 processes are running unconfined
- [x] AppArmor is loaded on mon
    - [x] 0 processes are running unconfined
- [x] Both servers are running grsec kernels
- [x] iptables rules loaded
- [x] OSSEC emails begin to flow after install
- [x] OSSEC emails are decrypted to correct key and I am able to decrypt them

**See notes in [QA Matrix](https://docs.google.com/spreadsheets/d/1ilHLJunjNSw57S4HcUYOB8TDtSAyMq6AXdPx8UfI4hM/edit#gid=1698153040)**

#### Command Line User Generation

- [x] Can successfully add admin user and login

### 2.0.2 release-specific changes

- [x] Kernel is updated to 5.4.136 [#6054](https://github.com/freedomofpress/securedrop/pull/6054)
  - [x] both app and mon servers are using the 5.4.136 grsec kernel 
  - [x] paxtest tests return expected results

**Comment by sssoleileraaa** [comment]
For prosperity, whenever a PR is opened (phase 1) in https://github.com/freedomofpress/securedrop-debian-packages-lfs with the branch name "release", the changes are pushed to https://apt-qa.freedom.press/. Then the package on https://apt-qa.freedom.press/ is signed with the prod key and we run through our Preflight Test plan (my results will be posted below). Once all our preflight testing comes back positive, the PR is merged (phase 2) which promotes the SAME package to be pushed to https://apt.freedom.press/.

## 2.0.2 QA Preflight Testing

### Environment

- Install target: nuc10-app/nuc8-mon
- Tails version: 4.21
- Test Scenario: "Clean install scenario" after modifying https://github.com/freedomofpress/securedrop/blob/058d5a0f71e3ffe9da027fafdb59263348ec9d7f/install_files/ansible-base/roles/install-fpf-repo/defaults/main.yml#L13 to point to `https://apt-qa.freedom.press/` instead of `https://apt.freedom.press/`
- SSH over Tor: no
- General notes: 2.0.1 -> 2.0.2

### Basic testing

- [x] Install or upgrade occurs without error
- [x] Source interface is available and version string indicates it is 2.0.2
- [x] A message can be successfully submitted

### Tails

- [x] The updater GUI appears on boot
- [x] The update successfully occurs to 2.0.2
- [x] After reboot, updater GUI no longer appears

**Comment by eloquence** [comment]
GUI update from 2.0.1->2.0.2 on Tails 4.21 worked without issues.
